internal/mux/mux_test.go

```diff
@@ -140,3 +140,17 @@ func TestNoReader(t *testing.T) {
 		t.Errorf("got: %v, want: %v", buf, make([]byte, len(buf)))
 	}
 }
+
+func TestEmptyBuffersDontBlock(t *testing.T) {
```

Comment:
@hajimehoshi This is the test I intended to test weak synchronization.

Reply:
Acknowledged